sample-service/src/main/java/uk/ac/ox/ndph/mts/sample_service/security/AADWebSecurityConfig.java

```diff
@@ -0,0 +1,49 @@
+package uk.ac.ox.ndph.mts.sample_service.security;
+
+import com.microsoft.azure.spring.autoconfigure.aad.AADAppRoleStatelessAuthenticationFilter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.authentication.HttpStatusEntryPoint;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+/**
+ * The security configuration class to support AAD with JWT (no sessions)
+ */
+@ConditionalOnProperty(prefix = "azure.activedirectory", value = "client-id")
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class AADWebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private AADAppRoleStatelessAuthenticationFilter aadAuthFilter;
+
+    /**
+     * Http security configuration
+     * @param http - http security
+     * @throws Exception - general exception
+     */
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        // disable csrf because we are using another token mechanism
+        http.csrf().disable(); //NOSONAR
+
+        // Do not create user sessions
+        http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.NEVER);
+
+        // This requires all requests to have a valid token
+        http.authorizeRequests()
+                .anyRequest().authenticated();
+
+        // This enables us to return appropriate http codes
+        http.exceptionHandling()
+                .authenticationEntryPoint(new HttpStatusEntryPoint(HttpStatus.UNAUTHORIZED));
+
+        http.addFilterBefore(aadAuthFilter, UsernamePasswordAuthenticationFilter.class);
```

Comment:
See comment above - do we need a custom AD filter - spring security already has oauth support built in - configured with something like:
  http.authorizeRequests().....
        .oauth2ResourceServer(oauth2 -> oauth2.jwt());

Reply:
Created a story to try and change that to use the Spring Security libraries: https://ndph-arts.atlassian.net/browse/ARTS-591